test/parallel/test-assert-strict-typecheck.js

```diff
@@ -0,0 +1,93 @@
+'use strict';
+require('../common');
+const assert = require('assert');
+
+// The following deepEqual tests might seem very weird.
+// They just describe what it is now.
+// That is why we discourage using deepEqual in our own tests.
+
+const date = new Date('2016');
+
+// See #10258
+function FakeDate() {}
+FakeDate.prototype = Date.prototype;
+const fake = new FakeDate();
+
+/* eslint-disable no-restricted-properties */
+// We are testing deepEqual!
+
+assert.doesNotThrow(() => assert.deepEqual(date, fake));
+assert.doesNotThrow(() => assert.deepEqual(fake, date));
+
+// For deepStrictEqual we check the runtime type,
+// then reveal the fakeness of the fake date
+assert.throws(() => assert.deepStrictEqual(date, fake),
+                    `AssertionError: ${date} deepStrictEqual Date { }`);
```

Comment:
Sorry for not reading the docs more thoroughly. I'll wrap them up in `new RegExp`, thanks!